Improve catalog error message for missing schemas

krleonid opened the pull request:
This PR improves catalog error messages for missing schemas.

## Changes
- Improve catalog error message for missing schemas

krleonid committed:
Improve catalog error message for missing schemas

When querying a table in a non-existent schema, the error message
now correctly indicates that the schema does not exist, rather than
saying the table does not exist.

The new error message format is:
'Catalog with name <schema> does not exist and there is no <schema> schema in the current catalog!'

This provides clearer feedback to users when they reference a schema
that doesn't exist, making it easier to identify and fix the issue.

Changes:
- Added logic in Catalog::TryLookupEntry to detect when a schema
  doesn't exist and throw an appropriate error
- Simplified code by removing redundant checks
- Added comprehensive tests for various scenarios including:
  - Non-existent schema without catalog specification
  - Non-existent schema with catalog specification
  - Non-existent table in existing schema (still shows table error)
  - Multiple catalog scenarios

krleonid committed:
Update test: rename board_21 to non_exist_schema for clarity

krleonid committed:
Fix schema error detection to only apply for table lookups

- Add check to only apply schema error logic when looking for TABLE_ENTRY
- This prevents incorrect schema errors for column.method calls like filename.replace()
- Update test expectations to match new accurate error messages
- All tests pass including dot_function_missing_error.test

krleonid committed:
Rename sqlite_scanner patch file from getdatainternal.patch to fix.patch

krleonid committed:
Merge remote-tracking branch 'origin/main' into improve_catalog_error_message

pdet commented on src/catalog/catalog.cpp:
```diff
@@ -909,6 +909,21 @@ CatalogEntryLookup Catalog::TryLookupEntry(CatalogEntryRetriever &retriever, con
+			auto except = CatalogException(
+			    schema_lookup.GetErrorContext(),
+			    "Catalog with name %s does not exist and there is no %s schema in the current catalog!", schema_name,
+			    schema_name);
```
Wouldn't  "Schema with name %s does not exist!" be a more clear error message?

pdet reviewed (dismissed):
Thanks for the PR, i had a small comment about it.

krleonid committed:
Change error message to show schema.table format for missing relations

Updated error message from 'Catalog with name X does not exist and there is no X schema in the current catalog!' to 'the relation "schema.table" does not exist' format. This provides clearer error messages showing the full qualified relation name.

krleonid committed:
Update error messages to show schema.table format for missing relations

- Updated catalog.cpp to generate error message in format 'the relation "schema.table" does not exist'
- Updated all test files to expect the new error message format
- Fixed remaining occurrences in test_schema.test and test_schema_dependency.test

krleonid committed:
Merge remote-tracking branch 'origin/main' into improve_catalog_error_message

krleonid committed:
Update sqlite_scanner patch file with new error message format

krleonid committed:
Update giant_query.test_slow to use new error message format

krleonid committed:
Fix relation name in sqlite_scanner patch (s1.test instead of s.test)

taniabogatsch commented on src/catalog/catalog.cpp:
```diff
@@ -909,6 +909,20 @@ CatalogEntryLookup Catalog::TryLookupEntry(CatalogEntryRetriever &retriever, con
+			EntryLookupInfo schema_lookup(CatalogType::SCHEMA_ENTRY, schema_name, lookup_info.GetErrorContext());
+			string relation_name = schema_name + "." + lookup_info.GetEntryName();
+			auto except =
+			    CatalogException(schema_lookup.GetErrorContext(), "the relation \"%s\" does not exist", relation_name);
```
But that caused the confusion in the first place, no? Since it was not actually the table that was missing?

taniabogatsch replied:
<img width="436" height="56" alt="Screenshot 2025-11-28 at 13 30 40" src="https://github.com/user-attachments/assets/a5c1ec29-312d-4a9a-a713-ebed612e84cf" />

So to confirm: looking at, e.g., this change. We want to stay with "table" and we want the error message to say: 
`Catalog Error: Table with name "s1.test` does not exist`?

taniabogatsch replied:
Isn't that still misleading because, like you said, it happens during schema lookup? Shouldn't we at least add something like `because schema "s1" does not exist`?

krleonid replied:
@Mytherin  Thank you for the review, 
I changed the error message in this PR only for the following case:

```
SELECT COUNT(*) FROM board_21.data
Catalog Error: Table with name data does not exist!
```
In this scenario, the schema board_21 is missing.
Keeping the message “Table with name board_21 does not exist” would be inaccurate, because as far as I understand, board_21 cannot represent a table here — it can only be a schema or a catalog.

Do you want to keep the same message, just with the correct board name?

krleonid committed:
Update error message to include schema information

Changed error message from 'the relation "schema.table" does not exist' to 'Table with name "schema.table" does not exist because schema "schema" does not exist.' to provide clearer information about why the table lookup failed.

krleonid committed:
Merge remote-tracking branch 'origin/main' into improve_catalog_error_message

krleonid committed:
Format CatalogException construction in TryLookupEntry

krleonid committed:
Update sqlite_scanner fix.patch for improved catalog error message

krleonid committed:
Merge remote-tracking branch 'upstream/main' into improve_catalog_error_message
